src/main/webapp/app/shared/util/core-submission/core-submission.spec.ts

```diff
@@ -0,0 +1,641 @@
+            tumor,
+            mutation,
+            ti: tiIr,
+            treatment,
+          },
+        ];
+      }),
+      [
+        'mutations/0/tumors/0/TIs/2/treatments/0',
+        {
+          ...baseExpectedArgs,
+          type: EvidenceEvidenceTypeEnum.StandardTherapeuticImplicationsForDrugSensitivity,
+          tumor,
+          mutation,
+          ti: tiSs,
+          treatment,
+        },
+      ],
+      [
+        'mutations/0/tumors/0/TIs/3/treatments/0',
+        {
+          ...baseExpectedArgs,
+          type: EvidenceEvidenceTypeEnum.StandardTherapeuticImplicationsForDrugResistance,
+          tumor,
+          mutation,
+          gene,
+          ti: tiSr,
+          treatment,
+        },
+      ],
+    ];
+    test.each(goodTests)('Path should parse as expected for path %s', (valuePath, args) => {
+      expect(pathToGetEvidenceArgs({ ...baseExpectedArgs, valuePath })).toEqual(args);
+    });
+    const badTests: [valuePath: string][] = [
+      ['bad/path'],
+      ['mutations/0/bad'],
+      ['mutations/300'],
+      ['mutations/bad/0/mutation_effect/'],
+      ['/'],
+      [''],
+    ];
+    test.each(badTests)('Should throw exception for path "%s"', (valuePath: string) => {
+      expect(() => pathToGetEvidenceArgs({ ...baseExpectedArgs, valuePath })).toThrow(Error);
+    });
+  });
+  describe('evidences mapping tests', () => {
+    type PathArgs = Parameters<typeof pathToGetEvidenceArgs>[0];
+    const baseArgs: Omit<PathArgs, 'valuePath' | 'gene'> = {
+      drugListRef: {},
+      entrezGeneId: Math.random(),
+      updateTime: new Date().getTime(),
+    };
+    const baseEvidence: Evidence = {
+      additionalInfo: null,
+      alterations: null,
+      articles: null,
+      description: null,
+      fdaLevel: null,
+      knownEffect: null,
+      levelOfEvidence: null,
+      liquidPropagationLevel: null,
+      solidPropagationLevel: null,
+      treatments: null,
+      cancerTypes: null,
+      excludedCancerTypes: null,
+      relevantCancerTypes: null,
+    };
+
+    const getTimeFromDateString = (s: string) => new Date(s).getTime();
+
+    const goodTests: [PathArgs, args: GetEvidenceRtn][] = [
+      [
+        {
+          ...baseArgs,
+          valuePath: 'summary',
+          gene: createMockGene({
+            name: 'GeneSummary',
+            summary: 'Gene Summary',
+            summary_uuid: 'd7ea4dc5-383d-40b8-a51f-725d25c087d2',
+            summary_review: createMockReview({
+              updateTime: getTimeFromDateString('2000-01-01'),
+            }),
+          }),
+        },
+        {
+          ['d7ea4dc5-383d-40b8-a51f-725d25c087d2']: {
+            ...baseEvidence,
+            evidenceType: EvidenceEvidenceTypeEnum.GeneSummary,
+            description: 'Gene Summary',
+            gene: {
+              entrezGeneId: baseArgs.entrezGeneId,
+              hugoSymbol: 'GeneSummary',
```

Comment:
Avoid the confusion, I would suggest to add a mock hugo symbol as well, similar to enterzGeneId. Currently it changes based on evidence type, but it should not.

Reply:
Also I don't think you need to overwrite gene